website_event_sale_hide_ticket/models/event_event.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+
+from odoo import api, models
+
+
+class Event(models.Model):
+    _inherit = "event.event"
+
+    @api.onchange("event_type_id")
+    def _onchange_type(self):
+        super()._onchange_type()
+
+        if self.event_type_id.use_ticketing:
+            for type_ticket in self.event_type_id.event_ticket_ids:
+                ticket = self.event_ticket_ids.filtered(
+                    lambda t: t.product_id == type_ticket.product_id
+                    and t.price == type_ticket.price
+                )
```

Comment:
Considering `super` "resets" the tickets from the event type, maybe it's possible to do something like this instead:

```py
def _onchange_type(self):
    res = super()._onchange_type()
    if self.event_type_id.use_ticketing:
        for ticket, type_ticket in zip(self.event_ticket_ids, self.event_type_id.event_ticket_ids):
            ticket.show_in_website = type_ticket.show_in_website
```

It'd avoid the product+price filtering that may not be accurate (nothing prevents the user to have two tickets with same price and product, only different seats limit / description / etc)


Unit tests would be good too :)